src/Microsoft.Azure.ServiceBus/Filters/CorrelationFilter.cs

```diff
@@ -0,0 +1,180 @@
+        /// <value>The session identifier.</value>
+        public string SessionId
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Session identifier to reply to.
+        /// </summary>
+        /// <value>The session identifier to reply to.</value>
+        public string ReplyToSessionId
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Content type of the message.
+        /// </summary>
+        /// <value>The content type of the message.</value>
+        public string ContentType
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Application specific properties of the message.
+        /// </summary>
+        /// <value>The application specific properties of the message.</value>
+        public IDictionary<string, object> Properties => this.properties ?? (this.properties = new PropertyDictionary());
+
+        /// <summary>
+        /// Converts the value of the current instance to its equivalent string representation.
+        /// </summary>
+        /// <returns>A string representation of the current instance.</returns>
+        public override string ToString()
+        {
+            StringBuilder builder = new StringBuilder();
+
+            builder.Append("CorrelationFilter: ");
+
+            bool firstExpression = true;
+
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.CorrelationId", this.CorrelationId);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.MessageId", this.MessageId);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.To", this.To);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.ReplyTo", this.ReplyTo);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.Label", this.Label);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.SessionId", this.SessionId);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.ReplyToSessionId", this.ReplyToSessionId);
+            this.AppendPropertyExpression(ref firstExpression, builder, "sys.ContentType", this.ContentType);
+
+            foreach (var pair in this.Properties)
+            {
+                string propertyName = pair.Key;
+                object propertyValue = pair.Value;
+
+                this.AppendPropertyExpression(ref firstExpression, builder, propertyName, propertyValue);
+            }
+
+            return builder.ToString();
+        }
+
+        void AppendPropertyExpression(ref bool firstExpression, StringBuilder builder, string propertyName, object value)
+        {
+            if (value != null)
+            {
+                if (firstExpression)
+                {
+                    firstExpression = false;
+                }
+                else
+                {
+                    builder.Append(" And ");
```

Comment:
Wonder if for consistency with [SQLFilter syntax](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-sql-filter), `AND` should be capitalized in the output.